core/templates/dev/head/pages/topic-editor-page/services/topic-editor-state.service.spec.ts

```diff
@@ -516,14 +526,14 @@ describe('Topic editor state service', function() {
   it('should be able to set a new topic rights with an in-place copy',
     function() {
       var previousTopicRights = TopicEditorStateService.getTopicRights();
-      var expectedTopicRights = TopicRightsObjectFactory.createFromBackendDict(
+      var expectedTopicRights = topicRightsObjectFactory.createFromBackendDict(
         secondTopicRightsObject);
       expect(previousTopicRights).not.toEqual(expectedTopicRights);
 
       TopicEditorStateService.setTopicRights(expectedTopicRights);
 
       var actualTopicRights = TopicEditorStateService.getTopicRights();
-      expect(actualTopicRights).toEqual(expectedTopicRights);
+      expect(angular.equals(actualTopicRights, expectedTopicRights)).toBe(true);
```

Comment:
Interesting. Why this change?

Reply:
This was done because of the unexpectedly failing expectation. Since `toEqual` does the same as `angular.equals` (comparing the contents rather than strict equality), @kevinlee12 suggested that we use `angular.equals`. This might be happening because previously they were just some dict but now after conversion to classes, they were erring.

Reply:
That's not really an explanation that provides insight. Why does angular.equals() work but .toEqual() not work? Is this a pattern that we should be using in the future, or does .toEqual() work in other cases?

Reply:
Ah there was some mixing of JS type and TS type functions that sent off `toEquals`'s alarm. Fixed it and done!